src/config.rs

```diff
@@ -52,10 +52,25 @@ macro_rules! create_config {
             $(pub $i: $ty),+
         }
 
+        #[derive(RustcDecodable, Clone)]
+        pub struct ParsedConfig {
+            $(pub $i: Option<$ty>),+
+        }
+
         impl Config {
+
+          fn fill_from_parsed_config(mut self, parsed: &ParsedConfig) -> Config {
+          $(
+              if let Some(val) = parsed.$i {
+                self.$i = val;
```

Comment:
yeah, this is atom not allowing per project configs and me switching constantly between different projects...fix is under way ;)